pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -741,6 +742,12 @@ func filterChainMatch(listener *listener.Listener, fc *listener.FilterChain, lp
 		}
 	}
 
+	if match.ApplicationProtocols != "" {
+		if fc.FilterChainMatch == nil || strings.Join(fc.FilterChainMatch.ApplicationProtocols, ",") != match.ApplicationProtocols {
+			return false
```

Comment:
Not sure about the semantics:
- Should it be AND or OR? 
- Does the order matter?

Reply:
The logic should match when `match.ApplicationProtocols` is subset of the FilterChainMatch.ApplicationProtocols`

Reply:
Fixed